tests/cases/expr/let.slint

```diff
@@ -0,0 +1,72 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+export component TestCase {
+    public function test_simple() -> int {
+        let a = 1;
+
+        return a;
+    }
+
+    public function test_shadowing() -> int {
+        let a = 1;
+        let a = a + 1;
+
+        return a;
+    }
+
+    public function test_shadowing_scope() -> int {
+        let a = 1;
+
+        if (true) {
+            let a = a + 1;
+
+            return a;
+        }
+
+        return a;
+    }
+
+    public function test_shadowing_different_types() -> string {
+        let a = 1;
+        let a: float = a + 1.0;
+        let a = @tr("{}", a);
+
+        return a;
+    }
+
+    public function test_type_annotation() -> int {
+        let a: int = 1;
+
+        return a;
+    }
+
+    public function test_type_annotation_conversion() -> int {
+        let a: int = 1.0;
+
+        return a;
+    }
```

Comment:
there could also be a `out property <bool> test : test_simple() == 1 && test_shadowing() == 2 && ....` for the interpreter test.